src/analyze/analyze-security.c

```diff
@@ -144,12 +77,18 @@ static void security_info_free(struct security_info *i) {
         free(i->device_policy);
 
         strv_free(i->supplementary_groups);
-        strv_free(i->system_call_architectures);
+        set_free(i->system_call_architectures);
+
+        hashmap_free(i->system_call_filter);
+}
 
-        set_free(i->system_call_filter);
+static SecurityInfo *security_info_free_free(SecurityInfo *info) {
+        return NULL;
```

Comment:
this should call security_info_free() and free()

Reply:
Like this?
  
        static SecurityInfo *security_info_free_free(SecurityInfo *info) {
                security_info_free(info);
                free(info);
        }

Reply:
yes, and also return NULL